langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiAiServicesWithJsonSchemaIT.java

```diff
@@ -0,0 +1,43 @@
+package dev.langchain4j.model.azure;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.service.AiServicesWithJsonSchemaIT;
+import org.junit.jupiter.api.AfterEach;
+
+import java.util.List;
+import java.util.Set;
+
+import static dev.langchain4j.model.chat.Capability.RESPONSE_FORMAT_JSON_SCHEMA;
+import static dev.langchain4j.model.chat.request.ResponseFormat.JSON;
+
+class AzureOpenAiAiServicesWithJsonSchemaIT extends AiServicesWithJsonSchemaIT {
+
+    @Override
+    protected List<ChatLanguageModel> models() {
+        return List.of(
+                AzureOpenAiChatModel.builder()
+                        .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
+                        .apiKey(System.getenv("AZURE_OPENAI_KEY"))
+                        .deploymentName("gpt-4o-mini")
+                        .supportedCapabilities(Set.of(RESPONSE_FORMAT_JSON_SCHEMA))
+                        .responseFormat(JSON)
```

Comment:
Hmm, ` .responseFormat(JSON)` seems redundant here.

` .supportedCapabilities(Set.of(RESPONSE_FORMAT_JSON_SCHEMA))` means "this model _can_ consume `JsonSchema` in the `ResponseFormat`. If `ChatRequest` will contain`ResponseFormat`/`JsonSchema`, it will work."

` .responseFormat(JSON)` means "always output JSON (without schema)"

".strictJsonSchema(true)" means "be strict with JSON schema"

These should work (currently they fails with NPEs):
```java
@Override
    protected List<ChatLanguageModel> models() {
        return List.of(
                AzureOpenAiChatModel.builder()
                        .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
                        .apiKey(System.getenv("AZURE_OPENAI_KEY"))
                        .deploymentName("gpt-4o-mini")
                        .supportedCapabilities(Set.of(RESPONSE_FORMAT_JSON_SCHEMA))
                        .strictJsonSchema(true)
                        .temperature(0.0)
                        .logRequestsAndResponses(true)
                        .build(),
                AzureOpenAiChatModel.builder()
                        .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
                        .apiKey(System.getenv("AZURE_OPENAI_KEY"))
                        .deploymentName("gpt-4o-mini")
                        .supportedCapabilities(Set.of(RESPONSE_FORMAT_JSON_SCHEMA))
                        .strictJsonSchema(false)
                        .temperature(0.0)
                        .logRequestsAndResponses(true)
                        .build()
        );
    }
```

Reply:
I've corrected this and modified the test to be like the one you coded above